common/djangoapps/student/views.py

```diff
@@ -660,7 +672,8 @@ def dashboard(request):
     # Find programs associated with courses being displayed. This information
     # is passed in the template context to allow rendering of program-related
     # information on the dashboard.
-    meter = programs_utils.ProgramProgressMeter(user, enrollments=course_enrollments)
+    # Use list() method for coping list because 'meter.engaged_programs' sorting enrollments by created date.
+    meter = programs_utils.ProgramProgressMeter(user, enrollments=list(course_enrollments))
```

Comment:
why do we need this change?

Reply:
Because, the method "openedx.core.djangoapps.programs.utils.ProgramProgressMeter.engaged_programs" has sorting enrollments by created date. This functionality is needed for this method. That's why I used list() for copy enrollments.
You can see that by this link: https://github.com/raccoongang/edx-platform/blob/ficus-rg/openedx/core/djangoapps/programs/utils.py#L201

Reply:
I mean that the `course_enrollments` had already been converted into the `list`, L620. Why is it not enough?

Reply:
I see what you mean.
I changed this line. I replaced list() to copy().